site/app/libraries/database/DatabaseQueries.php

```diff
@@ -5615,10 +5615,11 @@ private function getGradedGradeablesUserOrTeam(array $gradeables, $users, $teams
                     SELECT
                       json_agg(gcm_id) AS string_mark_id,
                       gc_id,
-                      gd_id
+                      gd_id,
+                      gcd_grader_id
                     FROM gradeable_component_mark_data
-                    GROUP BY gc_id, gd_id
-                  ) AS gcmd ON gcmd.gc_id=in_gcd.gc_id AND gcmd.gd_id=in_gcd.gd_id
+                    GROUP BY gc_id, gd_id, gcd_grader_id
+                  ) AS gcmd ON gcmd.gc_id=in_gcd.gc_id AND gcmd.gd_id=in_gcd.gd_id AND gcmd.gcd_grader_id=in_gcd.gcd_grader_id
```

Comment:
So that I understand this, for peer grading, each peer will submit their own gradeable_component_data row with its own gradeable_component_mark_data?